simple-service-webapp-rest/src/main/java/com/example/rest/AuthenticationEndpoint.java

```diff
@@ -38,11 +45,28 @@ private void authenticate(String username, String password) throws Exception {
         if(username.equals("abcde"))
             if(password.equals("12345"))
                 return;
-        else
-            throw new Exception();
+        else {
+
+                throw new ForbiddenException();
+        }
     }
 
-    private String issueToken(String username) {
-        return "token12";
+    private Token issueToken(String username) {
+
+        Token token = tokenRepository.containsValidToken(username);
+
+        if(token!=null){
+            //try if there are valid tokens
+            return token;
+        }
+        //else branch issue new token
+        token  = new Token();
+        Date date = new Date();
+        token.setToken(UUID.randomUUID() + username + date);
```

Comment:
Why not make a separate field for the username, for example, you might want to get the username later for the token object. It would be easier. Also don't worry about adding stuff the UUID.random, it is technically guaranteed to never return the same value again